### PR TITLE
[3.7] bpo-39401: Avoid unsafe DLL load on Windows 7 and earlier (GH-18231)

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue39401

Automerge-Triggered-By: @zooba

### CODE REVIEW

File: PC/getpathp.c [hunk 1]
```diff
@@ -245,7 +245,8 @@ static void
 join(wchar_t *buffer, const wchar_t *stuff)
 {
     if (_PathCchCombineEx_Initialized == 0) {
-        HMODULE pathapi = LoadLibraryW(L"api-ms-win-core-path-l1-1-0.dll");
+        HMODULE pathapi = LoadLibraryExW(L"api-ms-win-core-path-l1-1-0.dll", NULL,
```
Comment (anthonywee) [inline PC/getpathp.c:248]: Do we need to check that the `LOAD_LIBRARY_SEARCH_` flags are available on the machine? It looks like KB2533623 
 needs to be installed: https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw

Comment (anthonywee) [inline PC/getpathp.c:248]: Here's what I was thinking:
```suggestion
         _PathCchCombineEx = NULL;
        if (GetProcAddress(GetModuleHandleW(L"kernel32.dll"), "AddDllDirectory") != NULL) {
            /* Check that we can use the LOAD_LIBRARY_SEARCH_SYSTEM32 flag below by ensuring
               the AddDllDirectory method exists first */
            HMODULE pathapi = LoadLibraryExW(L"api-ms-win-core-path-l1-1-0.dll", NULL,
                                            LOAD_LIBRARY_SEARCH_SYSTEM32);
            if (pathapi) {
                _PathCchCombineEx = (PPathCchCombineEx)GetProcAddress(pathapi, "PathCchCombineEx");
            }
        }

```

Comment (anthonywee) [reply PC/getpathp.c:248]: (the suggestion isn't exactly correct since the GitHub suggestion was only for a single line)

Comment (zooba) [reply PC/getpathp.c:248]: The check is in the installer (I backported it for 3.7).

Considering Windows 7 is EOL, I don't think it's unreasonable to require the update.

Comment (anthonywee) [reply PC/getpathp.c:248]: Ah cool, just noticed that. Thanks!

**Pull request merged**